src/realm/views.hpp

```diff
@@ -56,20 +56,53 @@ class SortDescriptor {
     {
         return !m_columns.empty();
     }
-
-    // handover support
-    using HandoverPatch = std::unique_ptr<SortDescriptorHandoverPatch>;
-    static void generate_patch(SortDescriptor const&, HandoverPatch&);
-    static SortDescriptor create_from_and_consume_patch(HandoverPatch&, Table const&);
-
+    // returns whether this descriptor has any custom ascending/descending order
+    bool has_custom_order() const noexcept
+    {
+        return std::any_of(m_ascending.begin(), m_ascending.end(), [](bool b) { return !b; });
```

Comment:
I'm a little confused by what this check is trying to do. What does "custom" signify here? Aren't all sort orders custom (compared to the natural, unsorted order that the rows were returned in)? In what sense is a column being sorted as descending more custom than one that's sorted ascending?